virt-v2v/cold/entrypoint.go

```diff
@@ -0,0 +1,260 @@
+	disks, err := filepath.Glob(diskKind)
+	if err != nil {
+		fmt.Println("Error getting disks ", err)
+		return
+	}
+
+	for _, disk := range disks {
+		diskNum, err := strconv.Atoi(disk[num:])
+		if err != nil {
+			fmt.Println("Error geting disks names ", err)
+			return err
+		}
+		diskLink := fmt.Sprintf("%s/%s-sd%s", DIR, os.Getenv("V2V_vmName"), genName(diskNum+1))
+		diskImgPath := disk
+		if diskKind == FS {
+			diskImgPath = fmt.Sprintf("%s/disk.img", disk)
+		}
+		if err = os.Symlink(diskImgPath, diskLink); err != nil {
+			fmt.Println("Error creating disk link ", err)
+			return err
+		}
+	}
+	return
+}
+
+func executeVirtV2v(args []string) (err error) {
+	virtV2vCmd := exec.Command(args[0], args[1:]...)
+	virtV2vStdoutPipe, err := virtV2vCmd.StdoutPipe()
+	if err != nil {
+		fmt.Printf("Error setting up stdout pipe: %v\n", err)
+		return
+	}
+
+	tee := io.TeeReader(virtV2vStdoutPipe, os.Stdout)
+	virtV2vCmd.Stderr = os.Stderr
+
+	fmt.Println("exec ", virtV2vCmd)
+	if err = virtV2vCmd.Start(); err != nil {
+		fmt.Printf("Error executing command: %v\n", err)
+		return
+	}
+
+	virtV2vMonitorCmd := exec.Command("/usr/local/bin/virt-v2v-monitor")
+	virtV2vMonitorCmd.Stdin = tee
+	virtV2vMonitorCmd.Stdout = os.Stdout
+	virtV2vMonitorCmd.Stderr = os.Stderr
+
+	if err = virtV2vMonitorCmd.Start(); err != nil {
+		fmt.Printf("Error executing monitor command: %v\n", err)
+		return
+	}
+
+	if err = virtV2vCmd.Wait(); err != nil {
+		fmt.Printf("Error waiting for virt-v2v to finish: %v\n", err)
+		return
+	}
+	return
+}
+
+func getXMLFile(dir, fileExtension string) (string, error) {
+	files, err := filepath.Glob(filepath.Join(dir, "*."+fileExtension))
+	if err != nil {
+		return "", err
+	}
+	if len(files) > 0 {
+		return files[0], nil
+	}
+	return "", fmt.Errorf("XML file was not found.")
+}
+
+func xmlHandler(w http.ResponseWriter, r *http.Request) {
+	if xmlFilePath == "" {
+		fmt.Println("Error: XML file path is empty.")
+		http.Error(w, "XML file path is empty", http.StatusInternalServerError)
+		return
+	}
+
+	xmlData, err := os.ReadFile(xmlFilePath)
+	if err != nil {
+		fmt.Printf("Error reading XML file: %v\n", err)
+		http.Error(w, "Error reading XML file", http.StatusInternalServerError)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/xml")
+	_, err = w.Write(xmlData)
+	if err == nil {
+		w.WriteHeader(http.StatusOK)
+	} else {
+		fmt.Printf("Error writing response: %v\n", err)
+		http.Error(w, "Error writing response", http.StatusInternalServerError)
+	}
+
+}
+
+func shutdownHandler(w http.ResponseWriter, r *http.Request) {
+	fmt.Println("Shutdown request received. Shutting down server.")
+	err := server.Shutdown(context.Background())
+	if err != nil {
+		fmt.Printf("Error shutting down server: %v\n", err)
+	} else {
+		fmt.Println("Server shut down successfully.")
+	}
+}
+
+func isValidSource(source string) bool {
+	supportedSources := map[string]bool{
+		OVA:     true,
+		vSphere: true,
+	}
```

Comment:
that seems a bit cumbersome - why not to have something like the following then:
```golang
switch(source) {
    case OVA, vSphere:
        return true
    default:
        return false
}
```